src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx

```diff
@@ -19,6 +19,66 @@ Go here to get or create your [New Relic user key](https://one.newrelic.com/laun
 
 Once you have your user key, in **Observability** click **Connect to New Relic One**, then paste your user key and click **Connect**.
 
+## Monitor code-level metrics [#code-level]
+
+CodeStream's code-level metrics give you detailed insight into how your code is performing.
+
+![A screenshot of a Python method with average duration, throughput, and error rate telemetry.](./images/code-level-metrics.png "A Python method with average duration, throughput, and error rate metrics.")
+
+<figcaption>When relevant, CodeStream's code-level metrics are shown in a CodeLens above your Python methods.</figcaption>
+
+<Callout variant="caution">
+Code-level metrics are only available for Python code and [supported Python frameworks](/docs/apm/agents/python-agent/getting-started/instrumented-python-packages). These metrics are shown via CodeLens text, a line of text inserted above your instrumented method.
+
+Your Python application code also needs to be monitored by the Python APM agent and your repository associated with a New Relic service.
```

Comment:
There could be value in adding a link directly to the python agent here. I'm not entirely sure who the audience is here, but if it includes prospective NR users who currently only use codestream, a link to the python intro doc could onboard some people!

Reply:
Done!